src/plugins/orangehrmDashboardPlugin/Dao/EmployeeTimeAtWorkDao.php

```diff
@@ -32,11 +31,29 @@ class EmployeeTimeAtWorkDao extends BaseDao
      */
     public function getLatestAttendanceRecordByEmpNumber(int $empNumber): ?AttendanceRecord
     {
-        return $this->getRepository(AttendanceRecord::class)
-            ->findOneBy(
-                ['employee' => $empNumber],
-                ['id' => ListSorter::DESCENDING]
-            );
+        $latestPunchInUtcTime = $this->getLatestPunchInUtcTimeByEmpNumber($empNumber);
+        if (is_null($latestPunchInUtcTime)) {
+            return null;
+        }
+        $qb = $this->createQueryBuilder(AttendanceRecord::class, 'attendanceRecord');
+        $qb->andWhere('attendanceRecord.employee = :empNumber');
+        $qb->setParameter('empNumber', $empNumber);
+        $qb->andWhere('attendanceRecord.punchInUtcTime = :punchInUtcTime');
+        $qb->setParameter('punchInUtcTime', $latestPunchInUtcTime);
+        return $qb->getQuery()->getOneOrNullResult();
+    }
+
+    /**
+     * @param int $empNumber
+     * @return string|null
+     */
+    private function getLatestPunchInUtcTimeByEmpNumber(int $empNumber): ?string
+    {
+        $qb = $this->createQueryBuilder(AttendanceRecord::class, 'attendanceRecord')
+            ->select('MAX(attendanceRecord.punchInUtcTime)')
+            ->andWhere('attendanceRecord.employee = :empNumber')
+            ->setParameter('empNumber', $empNumber);
+        return $qb->getQuery()->getSingleScalarResult();
```

Comment:
```php
    public function getLatestAttendanceRecordByEmpNumber(int $empNumber): ?AttendanceRecord
    {
        $qb = $this->createQueryBuilder(AttendanceRecord::class, 'attendanceRecord')
            ->andWhere('attendanceRecord.employee = :empNumber')
            ->setParameter('empNumber', $empNumber)
            ->setMaxResults(1)
            ->addOrderBy('attendanceRecord.punchInUtcTime', ListSorter::DESCENDING);
        return $qb->getQuery()->getOneOrNullResult();
    }

    /**
     * @param int $empNumber
     * @return string|null
     */
    private function getLatestPunchInUtcTimeByEmpNumber(int $empNumber): ?string
    {
        // TODO:: remove this method
        $qb = $this->createQueryBuilder(AttendanceRecord::class, 'attendanceRecord')
            ->select('MAX(attendanceRecord.punchInUtcTime)')
            ->andWhere('attendanceRecord.employee = :empNumber')
            ->setParameter('empNumber', $empNumber);
        return $qb->getQuery()->getSingleScalarResult();
    }
```
try to refactor `getLatestAttendanceRecordByEmpNumber` and remove `getLatestPunchInUtcTimeByEmpNumber`, then check unit tests